### PR TITLE
MueLu: refactor of Phase3 aggregation, see issue #5838

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/muelu 

## Description
First implementation of Phase3 aggregation on device

## Motivation and Context
This work is part of a refactor of the setup of MueLu, the goal is to perform it on device if needed.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of #5838 
* Composed of 

## How Has This Been Tested?
Local build and tests have been performed.

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers.
- [x] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

**Comment by lucbv** [comment]
The current version needs to be tested on GPU so I will let the auto-tester take care of that.
Once the GPU build and tests are clean I will rebase the gold-files too.

**Comment by lucbv** [comment]
@csiefer2 @rstumin there are some changes compared to what is done in serial.
Let me know if you have any opinions on whether this version is reasonable or not?

File: packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase3Algorithm_kokkos_def.hpp [hunk 1]
```diff
@@ -94,129 +107,130 @@ namespace MueLu {
...
+                                   aggStat(nodeIdx)         = AGGREGATED;
+                                   procWinner(nodeIdx, 0)   = myRank;
+                                   vertex2AggId(nodeIdx, 0) = vertex2AggId(otherNodeIdx, 0);
+                                   Kokkos::atomic_decrement(&numNonAggregated());
+                                   return;
+                                 }
+                               }
+                             }
+
+                             // Total deperation!
```
Comment (csiefer2) [inline packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase3Algorithm_kokkos_def.hpp]: spelling

Comment (lucbv) [reply packages/muelu/src/Graph/UncoupledAggregation/MueLu_AggregationPhase3Algorithm_kokkos_def.hpp]: OK, I will fix it in the next wave of refactor as this should not impede the algorithm too much ; )

**Pull request merged**